src/groupByKey.ts

```diff
@@ -2,6 +2,22 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(object: T[], key: keyof T): GroupsMap<T> {
+  const finalObject: GroupsMap<T> = {};
+
+  object.forEach((el) => {
+    const keyValue = el[key];
+
+    if (keyValue !== undefined && keyValue !== null) {
+      const keyString = String(keyValue);
+
+      if (!finalObject.hasOwnProperty(keyString)) {
+        finalObject[keyString] = [];
+      }
+
+      finalObject[keyString].push(el);
+    }
```

Comment:
It's a good practice to ensure that the key used for grouping is not an empty string, as it might lead to unexpected behavior or logical errors in the grouping process.